src/hotspot/share/prims/jni.cpp

```diff
@@ -3465,9 +3465,11 @@ struct JNINativeInterface_* jni_functions_nocheck() {
 
 static void post_thread_start_event(const JavaThread* jt) {
   assert(jt != nullptr, "invariant");
+  // We hoist the read for the thread id to ensure the thread is assigned an id (lazy assignment).
```

Comment:
Suggestion:
```
// Ensure the thread is assigned an id (lazy assignment) even if the event isn't committed.
```

Reply:
Thanks David. I realized there is no need to move anything in jni.cpp, so I rolled them back. The change to the iterator will suffice.